src/pop3/POP3Server.ts

```diff
@@ -1,23 +1,30 @@
 import net from "net"
+import tls from "tls"
 import { readdirSync, readFileSync } from "fs"
 import User from "../models/User.js"
 import { createHash } from "node:crypto"
 import { readFile } from "fs/promises"
 import Mail from "../models/Mail.js"
 import Logger from "../Logger.js"
+import getConfig from "../config.js"
 
 const logger = new Logger("POP3", "YELLOW")
 
 export default class POP3Server {
 
 	server: net.Server
+	useTLS: boolean
 
-	constructor(port: number) {
-		this.server = net.createServer()
+	constructor(port: number, useTLS: boolean) {
+		this.useTLS = useTLS
+		this.server = useTLS ? tls.createServer({
+			key: readFileSync(getConfig("tls_key", "cert/privkey.pem")),
+			cert: readFileSync(getConfig("tls_cert", "cert/fullchain.pem"))
```

Comment:
1. use fs/promises
2. error handling - what if the paths provided aren't valid, the file does not exist or it is inaccessible?

Reply:
ok gonna look